validator/rpc/slashing.go

```diff
@@ -0,0 +1,108 @@
+// easy to migrate machines or eth2 clients.
+//
+// Steps:
+// 1. Open the validator database at the default location.
+// 2. Call the function which actually exports the data from
+//  the validator's db into an EIP standard slashing protection format.
+// 3. Format and save the JSON file to a user's specified output directory.
+func (s *Server) ExportSlashingProtection(ctx context.Context, _ *empty.Empty) (*pb.ExportSlashingProtectionResponse, error) {
+	var err error
+	// Default location
+	dataDir := s.walletDir
+
+	// Ensure that the validator.db is found under the specified dir or its subdirectories.
+	found, _, err := fileutil.RecursiveFileFind(kv.ProtectionDbFileName, dataDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error finding validator database at path %s", dataDir)
+	}
+	if !found {
+		return nil, errors.New("err finding validator database at pathh" + dataDir)
+	}
+
+	validatorDB, err := kv.NewKVStore(ctx, dataDir, &kv.Config{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not access validator database at path %s", dataDir)
+	}
+	defer func() {
+		if err := validatorDB.Close(); err != nil {
+			log.WithError(err).Errorf("Could not close validator DB")
+		}
+	}()
+	eipJSON, err := slashing.ExportStandardProtectionJSON(ctx, validatorDB)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not export slashing protection history")
+	}
+	encoded, err := json.MarshalIndent(eipJSON, "", "\t")
+	if err != nil {
+		return nil, errors.Wrap(err, "could not JSON marshal slashing protection history")
+	}
+
+	return &pb.ExportSlashingProtectionResponse{
+		File: string(encoded),
+	}, nil
+
+}
+
+// Import Slashing reads an input slashing protection EIP-3076
+// standard JSON string and attempts to insert its data into our validator DB.
+//
+// Steps:
+// 1. Open the validator database using the default datadir.
+// 2. Read the JSON string passed through rpc.
+// 3. Call the function which actually imports the data from
+// from the standard slashing protection JSON file into our database.
+func (s *Server) ImportSlashingProtection(ctx context.Context, req *pb.ImportSlashingProtectionRequest) (*emptypb.Empty, error) {
+	var err error
+	// Slashing Directory
+	dataDir := s.walletDir
+
+	// ensure that the validator.db is found under the specified dir or its subdirectories
+	found, _, err := fileutil.RecursiveFileFind(kv.ProtectionDbFileName, dataDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "err finding validator database at path %s", dataDir)
+	}
+	if !found {
+		return nil, errors.New("err finding validator database at path %s" + dataDir)
+	}
+
+	valDB, err := kv.NewKVStore(ctx, dataDir, &kv.Config{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not access validator database at path: %s", dataDir)
+	}
+	defer func() {
+		if err := valDB.Close(); err != nil {
+			log.WithError(err).Errorf("Could not close validator DB")
+		}
+	}()
```

Comment:
This whole section is not needed because we already have the database in `s.valDB`

Reply:
So we only need to check that s.valDB is not null?

Reply:
Also what guarantees it is in place.  For accounts we would check the server's walletInitialized, for the wallet RPC it was creating/recovering or modifying it.  This is unique in that we are assuming it is there.   So just the check?

Reply:
The validator cannot begin with a null database so it will definitely be defined. When you start the process, it has to open the database

Reply:
In the test case, we are controlling the creating of the server and i have a test where it is null. . Should i take it out , both from func and test func?